modules/alloy/src/main/ts/ephox/alloy/behaviour/blocking/BlockingApis.ts

```diff
@@ -0,0 +1,80 @@
+import { Optional, Optionals } from '@ephox/katamari';
+import { Attribute } from '@ephox/sugar';
+import * as BehaviourTypes from '../common/BehaviourTypes';
+import * as Behaviour from '../../api/behaviour/Behaviour';
+import { Keying } from '../../api/behaviour/Keying';
+import { Focusing } from '../../api/behaviour/Focusing';
+import { Replacing } from '../../api/behaviour/Replacing';
+import { AlloySpec } from '../../api/component/SpecTypes';
+import { AlloyComponent } from '../../api/component/ComponentApi';
+import * as GuiFactory from '../../api/component/GuiFactory';
+import { BlockingConfig } from './BlockingSchema';
+import { BlockingState } from './BlockingState';
+
+export interface BlockingApis extends BehaviourTypes.BehaviourApisRecord<BlockingConfig, BlockingState> {
```

Comment:
This should be in a separate types file for consistency.

- [ ] Move all blocking types to a `BlockingTypes.ts` file.

Reply:
I'd rather not: for the apis or the state I don't mind if the types are kept in separate files to the code, as any mismatches will get caught by tsc, but for the schema it's really important to me that the types are right next to the code because it would be really easy to let them get out of sync otherwise.

I have fixed the tree-shaking things, which (aside from consistency, which is pretty sketchy across the behaviours module) was I think the main problem here.

Reply:
I'm afraid I'm going to have to insist here, as I won't be approving this without this change. This type of thing needs to be done separately, as we want to keep things consistent. So if we're going to change the style it needs to be done for the other behaviours as well.

Reply:
Done